colossalai/engine/ophooks/zero_hook.py

```diff
@@ -0,0 +1,56 @@
+import torch
+from colossalai.registry import OPHOOKS
+from colossalai.zero.shard_utils import BaseShardStrategy
+
+from ._base_ophook import BaseOpHook
+
+
+@OPHOOKS.register_module
+class ZeroHook(BaseOpHook):
+    """
+    A hook to process sharded param for ZeRO method.
+    """
+
+    def __init__(self, shard_strategy: BaseShardStrategy):
+        super().__init__()
+        self.shard_strategy = shard_strategy
+
+    def pre_fwd_exec(self, module: torch.nn.Module, *args):
```

Comment:
add logic to deal with NestedNet.
see if elem in args is nn.Paramater.
gather the shared param?
and shard it afterwards?

Reply:
You just skip the nested model.

Reply:
OK